admin-dev/themes/new-theme/js/product-page/combination.js

```diff
@@ -0,0 +1,123 @@
+import $ from 'jquery';
+
+export default function() {
+  $(document).ready(function() {
+    let $jsCombinationsList = $('.js-combinations-list');
+    let idsProductAttribute = $jsCombinationsList.data('ids-product-attribute').toString().split(',');
+    let idsCount = idsProductAttribute.length;
+    let currentCount = 0;
+    let step = 5;
+
+    $.get($jsCombinationsList.attr('data-action-refresh-images') + '/' + $jsCombinationsList.data('id-product'))
+      .then(function(response) {
+        getCombinations(response);
+        $('#create-combinations').click(function() {
+          generate();
+        });
+      });
+
+    let combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
+
+    let getCombinations = (combinationsImages) => {
+      $.get(combinationUrl).then(function (resp) {
+        $('#loading-attribute').before(resp);
+        refreshImagesCombination(combinationsImages, idsProductAttribute.slice(currentCount, currentCount+step));
+        currentCount += step;
+        combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
+        if (currentCount <= idsCount) {
+          getCombinations(combinationsImages);
+        } else {
+          $('#combinations-bulk-form').removeClass('inactive');
+          $('#loading-attribute').fadeOut(1000).remove();
+        }
+      });
+    };
+  });
+
+  let refreshImagesCombination = (combinationsImages, idsProductAttribute) => {
+    $.each(idsProductAttribute, function (index, value) {
+      var $combinationElem = $('.combination[data="' + value + '"]');
+      var $imagesElem = $combinationElem.find('.images');
+      var $index = $combinationElem.attr('data-index');
+
+      $imagesElem.html('');
+
+      $.each(combinationsImages[value], function(key, image) {
+        $imagesElem.append('<div class="product-combination-image ' + (image.id_image_attr ? 'img-highlight' : '') + '">\
+                 <input type="checkbox" name="combination_' + $index + '[id_image_attr][]" value="' + image.id + '" ' + (image.id_image_attr ? 'checked="checked"' : '') + '>\
+                  <img src="' + image.base_image_url + '-small_default.' + image.format + '" alt="" />\
+                </div>');
+      });
+
+      $combinationElem.fadeIn(1000);
+    });
+
+    refreshDefaultImage();
+  };
+
+  let refreshDefaultImage = () => {
+    var productDefaultImageUrl = null;
+    var productCoverImageElem = $('#product-images-dropzone').find('.iscover');
```

Comment:
Yes, but I just move the code from old theme to the new one and I think it may be used a lot